source/matrix_free/mapping_info.inst.in

```diff
@@ -16,38 +16,126 @@
+    template void
+    internal::MatrixFreeFunctions::MappingInfo<deal_II_dimension, double, VectorizedArray<double, 2>>::
+      print_memory_consumption<ConditionalOStream>(ConditionalOStream &,
+                                                   const TaskInfo &) const;
+
+    template void internal::MatrixFreeFunctions::MappingInfo<
+      deal_II_dimension,
+      float, VectorizedArray<float, 4>>::print_memory_consumption<std::ostream>(std::ostream &,
+                                                     const TaskInfo &) const;
+    template void internal::MatrixFreeFunctions::MappingInfo<
+      deal_II_dimension,
+      float, VectorizedArray<float, 4>>::print_memory_consumption<ConditionalOStream>(ConditionalOStream &,
+                                                           const TaskInfo &)
+      const;
+#endif
+
+
+
+
+#  if DEAL_II_COMPILER_VECTORIZATION_LEVEL >= 2 && defined(__AVX__)
+    template struct internal::MatrixFreeFunctions::
+      MappingInfo<deal_II_dimension, double, VectorizedArray<double, 4>>;
+    template struct internal::MatrixFreeFunctions::
+      MappingInfo<deal_II_dimension, float, VectorizedArray<float, 8>>;
+
+    template struct internal::MatrixFreeFunctions::
+      MappingInfoStorage<deal_II_dimension, deal_II_dimension, double, VectorizedArray<double, 4> >;
+    template struct internal::MatrixFreeFunctions::
+      MappingInfoStorage<deal_II_dimension, deal_II_dimension, float, VectorizedArray<float, 8> >;
+#if deal_II_dimension > 1
+    template struct internal::MatrixFreeFunctions::
+      MappingInfoStorage<deal_II_dimension - 1, deal_II_dimension, double, VectorizedArray<double, 4> >;
+    template struct internal::MatrixFreeFunctions::
+      MappingInfoStorage<deal_II_dimension - 1, deal_II_dimension, float, VectorizedArray<float, 8> >;
+#endif
+
+    template void internal::MatrixFreeFunctions::MappingInfo<
+      deal_II_dimension,
+      double, VectorizedArray<double, 4>>::print_memory_consumption<std::ostream>(std::ostream &,
+                                                      const TaskInfo &) const;
+    template void
+    internal::MatrixFreeFunctions::MappingInfo<deal_II_dimension, double, VectorizedArray<double, 4>>::
+      print_memory_consumption<ConditionalOStream>(ConditionalOStream &,
+                                                   const TaskInfo &) const;
+
+    template void internal::MatrixFreeFunctions::MappingInfo<
+      deal_II_dimension,
+      float, VectorizedArray<float, 8>>::print_memory_consumption<std::ostream>(std::ostream &,
+                                                     const TaskInfo &) const;
+    template void internal::MatrixFreeFunctions::MappingInfo<
+      deal_II_dimension,
+      float, VectorizedArray<float, 8>>::print_memory_consumption<ConditionalOStream>(ConditionalOStream &,
+                                                           const TaskInfo &)
+      const;
+#endif
+
+
+
```

Comment:
Is there something missing for AVX-512?

Reply:
@kronbichler: regarding the files `dealii/source/matrix_free/matrix_free.inst.in` and `dealii/source/matrix_free/mapping_info.inst.in`. 

As you can see these files are a mess now. Explicit instantiation for float/double and width=1/2/4/8/16. I have inserted the empty lines to be distinguish  the instantiations. I will remove these as soon all tests have run.

Also I will add the AVX-512 instantiations asap.